src/datarobotx/idp/common/hashing.py

```diff
@@ -52,6 +54,10 @@ def get_hash(*args: Any, **kwargs: Any) -> str:
         elif arg is None:
             arg = int_to_bytes(_NONE_REPRESENTATION)
         elif isinstance(arg, str):
```

Comment:
I am not sure the assumption that "If is a string that corresponds to a file, than hash based on the file contents instead of the string value itself" is safe; an alternative here would be to modify the hash caller to pass in a list of Path objects to make the intent explicit vs dependent on the above assumption